src/Symfony/Component/Console/Input/ArgvInput.php

```diff
@@ -47,6 +47,10 @@ public function __construct(?array $argv = null, ?InputDefinition $definition =
     {
         $argv = $argv ?? $_SERVER['argv'] ?? [];
 
+        if (array_filter($argv, 'is_array')) {
+            throw new RuntimeException('Argument values expected to be all scalars, got array.');
+        }
```

Comment:
Catching arrays feels a bit arbitrary. What about objects? Resources?

Should we instead check for values that cannot be cast to string?

Reply:
Something like this (I'm not sure for the Stringable):
```suggestion
        foreach($argv as $arg) {
            if (!is_scalar($arg) && !$arg instanceof \Stringable) {
                throw new RuntimeException(sprintf('Argument values expected to be all scalars, got "%s".', get_debug_type($arg)));
            }
        }
```

Reply:
@GromNaN Applied your suggestion in the new PR https://github.com/symfony/symfony/pull/54576 targeting 7.1
Thank you